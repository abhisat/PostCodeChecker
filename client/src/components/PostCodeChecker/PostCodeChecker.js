import React from 'react';
import ReactDOM from 'react-dom';
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
import { FormControl } from '@material-ui/core';
import { InputLabel } from '@material-ui/core';
import { Input } from '@material-ui/core';
import styles from './style.module.css';
import { FormHelperText } from '@material-ui/core';

const theme = createMuiTheme({
  palette: {
    primary: { main: '#8BC3D1' },
  },
});
 
const PostCodeChecker = () => {
    return (
      <MuiThemeProvider theme={theme}>
        <FormControl className={styles.form}>
          <InputLabel htmlFor="postCode">Post Code</InputLabel>
          <Input id="postCode" aria-describedby="my-helper-text" />
          
          <InputLabel htmlFor="suburb">Suburb</InputLabel>
          <Input id="suburb" aria-describedby="my-helper-text" />
          
          <InputLabel htmlFor="state">State</InputLabel>
          <Input id="state" aria-describedby="my-helper-text" />
          
        </FormControl>
      </MuiThemeProvider>
    )
};

export { PostCodeChecker };
