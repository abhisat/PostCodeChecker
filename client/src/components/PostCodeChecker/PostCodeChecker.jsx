import React from 'react';
import ReactDOM from 'react-dom';
import { FormControl } from '@material-ui/core';
import styles from './style.module.css';
import { TextField } from '@material-ui/core';
import { Button } from '@material-ui/core';
import { Typography } from '@material-ui/core';
import { Select } from '@material-ui/core';
import { MenuItem } from '@material-ui/core';
 
const PostCodeChecker = (props) => {

  const { 
    postCode,
    suburb,
    region,
    onPostCodeChange,
    onSuburbChange,
    onRegionChange,
    onSubmit
   } = props;

  return (
    <div className={styles.formContainer}>
      <form onSubmit={onSubmit}>
        <FormControl className={styles.form} onSubmit={onSubmit}>
          <Typography className={styles.formHeader}>
            Find a Post Code
          </Typography>
          <TextField
            id="standard-name"
            label="Post Code"
            className={styles.textField}
            value={postCode}
            onChange={onPostCodeChange}
            margin="normal"
            type="text"
          />

          <TextField
            id="standard-name"
            label="Suburb"
            className={styles.textField}
            value={suburb}
            onChange={onSuburbChange}
            margin="normal"
            type="text"
          />
          <Select
            value={region}
            onChange={onRegionChange}
            className={styles.dropDown}
            displayEmpty
          >
            <MenuItem value="" disabled>
              <em>State</em>
            </MenuItem>
            <MenuItem value={"NSW"}>NSW</MenuItem>
            <MenuItem value={"VIC"}>VIC</MenuItem>
            <MenuItem value={"SA"}>SA</MenuItem>
            <MenuItem value={"WA"}>WA</MenuItem>
            <MenuItem value={"NT"}>NT</MenuItem>
            <MenuItem value={"TAS"}>TAS</MenuItem>
          </Select>
          <Button 
            variant="contained" 
            color="primary" 
            className={styles.button}
            type="submit"
            value="Submit"
          > 
            Check Address
          </Button>
        </FormControl>
      </form>
    </div>
  )
};

export { PostCodeChecker };
