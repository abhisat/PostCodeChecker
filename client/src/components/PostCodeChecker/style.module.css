.formContainer {
    width: 100%;
    height: 100%;
}

.form {
    width: 100%;
    height: 100%;
}

.formHeader {
    margin: 0 auto !important;
    color: #0075bf !important;
    font-size: 2rem !important;
}

.textField {
    margin: 0 auto !important;
    width: 30%;
}

.button {
    margin: 0 auto !important;
    margin-top: 15px !important;
    background-color: #0075bf !important;
}

.dropDown {
    margin: 0 auto !important;
    margin-top: 15px !important;
    width: 30%;
}