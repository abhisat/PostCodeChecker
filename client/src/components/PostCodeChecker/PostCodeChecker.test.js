import React from "react";
import { PostCodeChecker } from "./PostCodeChecker";
import Enzyme, { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EnzymeAdapter from 'enzyme-adapter-react-16';

Enzyme.configure({
  adapter: new EnzymeAdapter(),
});

const handlePostCodeChange = jest.fn()
const handleSuburbChange = jest.fn()
const handleRegionChange = jest.fn()
const handleSubmit = jest.fn()

const props = {
  postCode: "",
  suburb: "",
  region: "",
  onPostCodeChange: handlePostCodeChange,
  onSuburbChange: handleSuburbChange,
  onRegionChange: handleRegionChange,
  onSubmit: handleSubmit
}

const mountPostCodeChecker = (props = {}) => {
    const wrapper = shallow(<PostCodeChecker {...props} />);
    return wrapper;
};

describe("<PostCodeChecker />", () => {
  it("Matches the snapshot", () => {
    const PostCodeChecker = mountPostCodeChecker(props);
    expect(toJson(PostCodeChecker)).toMatchSnapshot();
  });

  it('handles postcode input', () => {
    const PostCodeChecker = mountPostCodeChecker(props);
    PostCodeChecker.find('#postCode').simulate('change');
    expect(handlePostCodeChange.mock.calls.length).toBe(1);
  });

  it('handles suburb input', () => {
    const PostCodeChecker = mountPostCodeChecker(props);
    PostCodeChecker.find('#suburb').simulate('change');
    expect(handleSuburbChange.mock.calls.length).toBe(1);
  });

  it('handles state input', () => {
    const PostCodeChecker = mountPostCodeChecker(props);
    PostCodeChecker.find('#region').simulate('change');
    expect(handleRegionChange.mock.calls.length).toBe(1);
  });

  it('handles submit', () => {
    const PostCodeChecker = mountPostCodeChecker(props);
    PostCodeChecker.find('form').simulate('submit');
    expect(handleSubmit.mock.calls.length).toBe(1);
  });
});