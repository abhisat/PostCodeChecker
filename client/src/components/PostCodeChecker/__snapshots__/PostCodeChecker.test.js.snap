// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCodeChecker /> Matches the snapshot 1`] = `
<div
  className="formContainer"
>
  <form
    onSubmit={[MockFunction]}
  >
    <WithStyles(ForwardRef(FormControl))
      className="form"
    >
      <WithStyles(ForwardRef(Typography))
        className="formHeader"
      >
        Find a Post Code
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(TextField))
        className="textField"
        id="postCode"
        label="Post Code"
        margin="normal"
        onChange={[MockFunction]}
        type="text"
        value=""
      />
      <WithStyles(ForwardRef(TextField))
        className="textField"
        id="suburb"
        label="Suburb"
        margin="normal"
        onChange={[MockFunction]}
        type="text"
        value=""
      />
      <WithStyles(ForwardRef(Select))
        className="dropDown"
        displayEmpty={true}
        id="region"
        onChange={[MockFunction]}
        value=""
      >
        <WithStyles(ForwardRef(MenuItem))
          disabled={true}
          value=""
        >
          <em>
            State
          </em>
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          value="NSW"
        >
          NSW
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          value="VIC"
        >
          VIC
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          value="SA"
        >
          SA
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          value="WA"
        >
          WA
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          value="NT"
        >
          NT
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          value="TAS"
        >
          TAS
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
      <WithStyles(ForwardRef(Button))
        className="button"
        color="primary"
        id="submit"
        type="submit"
        value="Submit"
        variant="contained"
      >
        Check Address
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(FormControl))>
  </form>
</div>
`;
